solutions/ac-MultipleOf2019.mdx

```diff
@@ -12,17 +12,40 @@ First of all, in order to store sections of the string, we can make a suffix arr
 
 $${S_i = s_n + 10*s_{n-1} +100*s_{n-2} + ... + {10^{n-i}} * s_{i}}$$
 
-For example, in the string "1234", the suffix array would be
+For example, in the string $$1234$$ and the modulus of $$3$$, the suffix array would be
 
-[1234, 234, 34, 4]
+$$[1234, 234, 34, 4, 0]$$
 
-This is useful because we can query any substring in $\mathcal{O}(1)$ time. So if $${S_0} = 1234 $$ and $${S_3=4}$$, this means $${S_0}-{S_3} = 123 * {10^1}$$ or $$1230$$. 
+**Keep in mind, that an empty suffix with is also considered.**
 
-It also means that if both $${S_0}(mod$$ $$M)=$$ $${S_3}(mod$$ $$M)$$, subtracting would lead to $${S_0}(mod$$ $$M)-$$ $${S_3}(mod$$ $$M)=0$$, meaning a substring that makes a number that is divisible by $$M$$ or $$2019$$. 
+This is useful because we can access any substring in $\mathcal{O}(1)$ time through a range query. So if $${S_0} = 1234 $$ and $${S_3=4}$$, this means $${S_0}-{S_3} = 123 * {10^1}$$ or $$1230$$. 
+
+It also means that if both $${S_0}(mod$$ $$M)=$$ $${S_3}(mod$$ $$M)$$ (both suffixes have the same remainders), subtracting would lead to $${S_0}(mod$$ $$M)-$$ $${S_3}(mod$$ $$M)=0$$, meaning a substring that makes a number that is divisible by $$M$$ or $$2019$$. 
+
+So with $$1234$$, $${S_0-{S_3}=1234(mod}$$ $$3)-4(mod$$ $$3)=1-1=0$$. Because they substract to zero, it means the substring from index $$0$$ to $$3$$ ($$3$$ is exclusive) leads to a number divisible by the modulus. (The substring is "$$123$$" which is divisble by 3). 
+
+However, we would still need to go through every pair of possible suffixes (i, j). 
+
+To take it one step further, **we do not need to know the specific indicies, we just need to know the amount of times a certain remainder occurs.** 
+
+Thus, we can use $${N \choose K}$$ formula, and choose 2 numbers that are divisible by the same reminder of 2019, where $$N=$$ the amount of times a certain remainder occurs, and $$K=$$ 2 (because we are picking 2 indicies $$i$$ and $$j$$). 
```

Comment:
> Thus, we can use $${N \choose K}$$ formula

again, what

Reply:
I mean this -> nCk=n!/(k!(n-k)!)

Does it sound awkward?

Reply:
yeah, uh... it's not a formula, is it?

Reply:
I see, I'll change that

Reply:
changed